VirtoCommerce.CustomerModule.Data/Handlers/LogChangesMemberChangedEventHandler.cs

```diff
@@ -0,0 +1,120 @@
+            _observedProperties = observedProperties;
+        }
+
+        public virtual Task Handle(MemberChangedEvent message)
+        {
+            //Log audit records only for contacts members
+            var operationLogs = message.ChangedEntries.Where(x => x.EntryState == EntryState.Modified && x.OldEntry is Contact)
+                                        .SelectMany(GetChangedEntryOperationLogs)
+                                        .ToArray();
+            if (operationLogs.Any())
+            {
+                _changeLogService.SaveChanges(operationLogs);
+            }
+            return Task.CompletedTask;
+        }
+
+        protected virtual IEnumerable<OperationLog> GetChangedEntryOperationLogs(GenericChangedEntry<Member> changedEntry)
+        {
+            var result = new List<string>();
+
+            var oldContact = changedEntry.OldEntry as Contact;
+            var changedContact = changedEntry.NewEntry as Contact;
+
+            var diff = Comparer.Compare(oldContact, changedContact);
+            //First, we detect changes with simple properties of the contact object.
+            var observedDifferences = diff.Join(_observedProperties, x => x.Name.ToLowerInvariant(), x => x.ToLowerInvariant(), (x, y) => x).ToArray();
+            var anonymousDiffComparer = AnonymousComparer.Create((Difference x) => string.Join(":", x.Name, x.NewValue, x.OldValue));
+            foreach (var difference in observedDifferences.Distinct(anonymousDiffComparer))
+            {
+                result.Add(string.Format(MemberResources.MemberPropertyChanged, difference.Name, difference.OldValue, difference.NewValue));
+            }
+            //Second, detect changes for contact dependencies
+            result.AddRange(GetContactDependenciesChanges(oldContact, changedContact));
+
+            return result.Select(x => GetLogRecord(changedContact, x));
+        }
+
+        protected virtual IEnumerable<string> GetContactDependenciesChanges(Contact originalContact, Contact modifiedContact)
+        {
+            return DetectCollectionsChanges(originalContact.Addresses, modifiedContact.Addresses, "Address")
+                    .Union(DetectCollectionsChanges(originalContact.Emails, modifiedContact.Emails, "Email"))
+                    .Union(DetectCollectionsChanges(originalContact.Phones, modifiedContact.Phones, "Phone"));
+        }
+
+        protected static IEnumerable<string> DetectCollectionsChanges<T>(IEnumerable<T> originalCollection, IEnumerable<T> modifiedCollection, string resourceKeyPrefix)
+        {
+            var result = new List<string>();
+            if (originalCollection != null && modifiedCollection != null)
+            {
+                void CompareAction(EntryState state, T source, T target)
+                {
+                    //Do not log unchanged values.
+                    if (state != EntryState.Modified || source?.ToString() != target?.ToString())
```

Comment:
Isn't this check done in [`CompareTo`](https://github.com/VirtoCommerce/vc-platform/blob/master/VirtoCommerce.Platform.Core/Common/CollectionExtensions.cs#L121-L142) method?
It doesn't call this action if there are no changes?